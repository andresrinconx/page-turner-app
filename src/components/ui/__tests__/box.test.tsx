import { render } from "@testing-library/react-native";
import { View } from "react-native";
import Box from "../box";

describe("Box", () => {
  it("renders correctly with default props", () => {
    const { getByTestId } = render(<Box testID="box" />);
    const box = getByTestId("box");
    expect(box).toBeTruthy();
    expect(box.type).toBe("View");
  });

  it("applies backgroundColor correctly", () => {
    const backgroundColor = "#FF0000";
    const { getByTestId } = render(
      <Box testID="box" backgroundColor={backgroundColor} />,
    );
    const box = getByTestId("box");
    expect(box.props.style).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          backgroundColor,
        }),
      ]),
    );
  });

  it("applies custom styles", () => {
    const customStyle = { padding: 10 };
    const { getByTestId } = render(<Box testID="box" style={customStyle} />);
    const box = getByTestId("box");
    expect(box.props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(customStyle)]),
    );
  });

  it("renders children correctly", () => {
    const testId = "test-box";
    const childTestId = "child-element";
    const { getByTestId } = render(
      <Box testID={testId}>
        <View testID={childTestId}>
          <View>Test Content</View>
        </View>
      </Box>,
    );

    const box = getByTestId(testId);
    const child = getByTestId(childTestId);

    expect(box).toBeTruthy();
    expect(child).toBeTruthy();
    expect(box.props.children).toBeTruthy();
  });
});
