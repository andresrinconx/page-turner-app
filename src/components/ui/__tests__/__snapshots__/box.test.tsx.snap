// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FF0000",
      },
      {
        "padding": 10,
      },
    ]
  }
>
  Test Content
</View>
`;
